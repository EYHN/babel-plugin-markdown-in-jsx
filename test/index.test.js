const babel = require('@babel/core');
const babelEnv = require('@babel/preset-env');
const babelReact = require('@babel/preset-react');
const plugin = require('../');

const transform = (code, option) => {
  const module = `
    const React = require('react');
    const Markdown = require('Markdown/component');
    function TestComponent() {
      ${code};
    }`;
  return babel.transform(module, {
    presets: [babelEnv, babelReact],
    plugins: [[plugin, {package: 'Markdown', ...option}]]
  }).code;
};

test('basic usage', () => {
  const renderBody = `return (
    <Markdown>
      # Title

      This is **bold.**

      Here is a [link](/some/url).
    </Markdown>
  )`;

  expect(transform(renderBody)).toMatchSnapshot();
});

test('call default', () => {
  const renderBody = `return (
    <Markdown.default>
      # Title

      This is **bold.**

      Here is a [link](/some/url).
    </Markdown.default>
  )`;

  expect(transform(renderBody)).toMatchSnapshot();
});

test('import declaration', () => {
  const renderBody = `import * as React from 'react';
import Markdown from 'Markdown/component';
function TestComponent() {
  return (
    <Markdown>
      # Title

      This is **bold.**

      Here is a [link](/some/url).
    </Markdown>
  )
}`;

  const code = babel.transform(renderBody, {
    presets: [babelEnv, babelReact],
    plugins: [[plugin, {package: 'Markdown'}]]
  }).code

  expect(code).toMatchSnapshot();
});

test('basic usage inline', () => {
  const renderBody = `return (
    <div>
      Here is <Markdown inline>_interpolated_ markdown</Markdown>
    </div>
  )`;

  expect(transform(renderBody)).toMatchSnapshot();
});

test('nested jsx', () => {
  const renderBody = `var number = 4;
    return (
      <Markdown>
        Here is a number: <span style={{ fontWeight: 'bold' }}>{number}</span>. Here it is again: {number}.
      </Markdown>
    )`;

  expect(transform(renderBody)).toMatchSnapshot();
});

test('extra props to element', () => {
  const renderBody = `return (
    <p>
      Here is <Markdown inline style={{ background: '#eee' }}>_interpolated_ markdown</Markdown>
    </p>
  )`;

  expect(transform(renderBody)).toMatchSnapshot();
});

test('nest Markdown components', () => {
  const renderBody = `return (
    <Markdown>
      **markdown**
      <div>
        **no markdown**
      </div>
      <Markdown>
        **markdown**
      </Markdown>
      {/*common comments*/}
    </Markdown>
  )`;

  expect(transform(renderBody)).toMatchSnapshot();
});

test('bulitin component proxy', () => {
  const ReactDOMServer = require('react-dom/server');
  const React = require('react');

  const proxy = require('../component/proxy');

  const html = ReactDOMServer.renderToString(<div>
      <proxy.div>123</proxy.div>
      <proxy.Provider value={{p: ({children}) => <p>123{children}</p>}}>
        <proxy.p>abc</proxy.p>
      </proxy.Provider>
      123
    </div>);

  expect(html).toMatchSnapshot();
});

test('transform to html with proxy', () => {
  const renderBody = `return (
    <Markdown>
      **markdown**
      <div>
        **no markdown**
      </div>
      <Markdown>
        123
        **markdown**
      </Markdown>
      {/*common comments*/}
    </Markdown>
  )`;

  expect(transform(renderBody, {proxy: true})).toMatchSnapshot();
});
