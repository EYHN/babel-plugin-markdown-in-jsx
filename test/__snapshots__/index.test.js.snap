// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`basic usage 1`] = `
"\\"use strict\\";

var React = require('react');

function TestComponent() {
  return React.createElement(\\"div\\", null, React.createElement(\\"h1\\", null, \\"Title\\"), React.createElement(\\"p\\", null, \\"This is \\", React.createElement(\\"strong\\", null, \\"bold.\\")), React.createElement(\\"p\\", null, \\"Here is a \\", React.createElement(\\"a\\", {
    href: \\"/some/url\\"
  }, \\"link\\"), \\".\\"));
}"
`;

exports[`basic usage inline 1`] = `
"\\"use strict\\";

var React = require('react');

function TestComponent() {
  return React.createElement(\\"div\\", null, \\"Here is \\", React.createElement(\\"span\\", null, React.createElement(\\"em\\", null, \\"interpolated\\"), \\" markdown\\"));
}"
`;

exports[`bulitin component proxy 1`] = `"<div data-reactroot=\\"\\"><div>123</div><p>123<!-- -->abc</p>123</div>"`;

exports[`call default 1`] = `
"\\"use strict\\";

var React = require('react');

function TestComponent() {
  return React.createElement(\\"div\\", null, React.createElement(\\"h1\\", null, \\"Title\\"), React.createElement(\\"p\\", null, \\"This is \\", React.createElement(\\"strong\\", null, \\"bold.\\")), React.createElement(\\"p\\", null, \\"Here is a \\", React.createElement(\\"a\\", {
    href: \\"/some/url\\"
  }, \\"link\\"), \\".\\"));
}"
`;

exports[`extra props to element 1`] = `
"\\"use strict\\";

var React = require('react');

function TestComponent() {
  return React.createElement(\\"p\\", null, \\"Here is \\", React.createElement(\\"span\\", {
    style: {
      background: '#eee'
    }
  }, React.createElement(\\"em\\", null, \\"interpolated\\"), \\" markdown\\"));
}"
`;

exports[`import declaration 1`] = `
"import * as React from 'react';

function TestComponent() {
  return React.createElement(\\"div\\", null, React.createElement(\\"h1\\", null, \\"Title\\"), React.createElement(\\"p\\", null, \\"This is \\", React.createElement(\\"strong\\", null, \\"bold.\\")), React.createElement(\\"p\\", null, \\"Here is a \\", React.createElement(\\"a\\", {
    href: \\"/some/url\\"
  }, \\"link\\"), \\".\\"));
}"
`;

exports[`nest Markdown components 1`] = `
"\\"use strict\\";

var React = require('react');

function TestComponent() {
  return React.createElement(\\"div\\", null, React.createElement(\\"p\\", null, React.createElement(\\"strong\\", null, \\"markdown\\"), React.createElement(\\"div\\", null, \\"**no markdown**\\"), React.createElement(\\"div\\", null, React.createElement(\\"p\\", null, React.createElement(\\"strong\\", null, \\"markdown\\")))));
}"
`;

exports[`nested jsx 1`] = `
"\\"use strict\\";

var React = require('react');

function TestComponent() {
  var number = 4;
  return React.createElement(\\"div\\", null, React.createElement(\\"p\\", null, \\"Here is a number: \\", React.createElement(\\"span\\", {
    style: {
      fontWeight: 'bold'
    }
  }, number), \\". Here it is again: \\", number, \\".\\"));
}"
`;

exports[`transform to html with proxy 1`] = `
"\\"use strict\\";

var P = require(\\"Markdown/component/proxy\\");

var React = require('react');

function TestComponent() {
  return React.createElement(P.div, null, React.createElement(P.p, null, React.createElement(P.strong, null, \\"markdown\\"), React.createElement(\\"div\\", null, \\"**no markdown**\\"), React.createElement(P.div, null, React.createElement(P.p, null, \\"123\\", React.createElement(P.strong, null, \\"markdown\\")))));
}"
`;

exports[`transform to html with proxy esmodules 1`] = `
"import * as P from \\"Markdown/component/proxy\\";
import * as React from 'react';

function TestComponent() {
  return React.createElement(P.div, null, React.createElement(P.h1, null, \\"Title\\"), React.createElement(P.p, null, \\"This is \\", React.createElement(P.strong, null, \\"bold.\\")), React.createElement(P.p, null, \\"Here is a \\", React.createElement(P.a, {
    href: \\"/some/url\\"
  }, \\"link\\"), \\".\\"));
}"
`;

exports[`with variable img href 1`] = `
"\\"use strict\\";

var React = require('react');

function TestComponent() {
  var h = 'http://example.com';
  return React.createElement(\\"div\\", null, React.createElement(\\"h1\\", null, \\"Title\\"), React.createElement(\\"p\\", null, React.createElement(\\"img\\", {
    src: \\"123\\" + h + \\"\\",
    alt: true
  })));
}"
`;
